### PR TITLE
Add material supplier management with validation

### PR DESCRIPTION
**Pull request by mnindrazaka**
## Summary
This PR adds functionality to manage material suppliers with comprehensive validation. It introduces a new `SetMaterialSuppliers` method that allows setting suppliers for materials with strict validation rules for different purchase types (online, offline, delivery).

## Key Changes

- **Material Supplier Management**: Added `SetMaterialSuppliers` usecase method with validation for:
  - Online purchases: requires valid HTTP(S) URLs (max 2048 chars)
  - Offline/Delivery purchases: must have empty purchase URLs
  - Supplier existence verification before assignment

- **Database Layer**: 
  - Implemented `ReplaceSuppliers` repository method that intelligently manages supplier associations (creates new, updates existing, soft-deletes removed)
  - Enhanced `GetMaterialById` to preload supplier relationships
  - Added `SoftDeleteMaterialSuppliersBySupplierId` for cascading deletes when suppliers are deleted

- **Supplier Deletion Cascade**: Modified `DeleteSupplierById` to use transactions and automatically soft-delete associated material suppliers

- **Data Models**:
  - Added `Suppliers` field to Material entity
  - Added transformer methods for MaterialSupplier domain conversion
  - Updated repository interfaces with new methods

- **Testing**:
  - Added comprehensive test suite for `SetMaterialSuppliers` with 13 test cases covering success scenarios and validation edge cases
  - Updated existing tests to pass new SupplierRepository dependency
  - Added helper function `newMaterialUsecase` to reduce test boilerplate
  - Updated supplier deletion tests to verify transaction behavior

## Notable Implementation Details

- URL validation uses Go's `net/url` package with scheme and host verification
- Supplier replacement uses a lookup map to efficiently determine which records to create, update, or soft-delete
- Transaction support ensures data consistency when cascading supplier deletions
- All soft-deletes preserve audit trails via `deleted_at` timestamps

https://claude.ai/code/session_01Bhuaty1vMTfgVkc7E2wqDi

### CODE REVIEW

File: apps/api/domain/material_usecase.go [hunk 1]
```diff
@@ -87,3 +89,38 @@ func (usecase MaterialUsecase) UpdateMaterialById(ctx context.Context, material
 func (usecase MaterialUsecase) DeleteMaterialById(ctx context.Context, id int64) *Error {
 	return usecase.repository.DeleteMaterialById(ctx, id)
 }
+
+func (usecase MaterialUsecase) SetMaterialSuppliers(ctx context.Context, materialId int64, payload []MaterialSupplier) *Error {
```
Comment (mnindrazaka) [inline apps/api/domain/material_usecase.go]: Why we use seperate usecase for setMaterialSuppliers ? 
Why not set it when creating or updating the material ? So we don't need to create a new usecase and repo

File: apps/api/domain/material_usecase.go [hunk 2]
```diff
@@ -87,3 +89,38 @@ func (usecase MaterialUsecase) UpdateMaterialById(ctx context.Context, material
...
+func isValidHttpUrl(raw string) bool {
+	if raw == "" || len(raw) > 2048 {
+		return false
+	}
+	parsed, err := url.Parse(raw)
+	if err != nil {
+		return false
+	}
+	return (parsed.Scheme == "http" || parsed.Scheme == "https") && parsed.Host != ""
+}
```
Comment (mnindrazaka) [inline apps/api/domain/material_usecase.go]: Can we move this to other place ? this look like general or global utils that is not belongs to any domain ?

File: apps/api/domain/material_repository.go [hunk 3]
```diff
@@ -15,4 +15,5 @@ type MaterialRepository interface {
 	CreateMaterial(ctx context.Context, material Material) (Material, *Error)
 	UpdateMaterialById(ctx context.Context, material Material, id int64) (Material, *Error)
 	DeleteMaterialById(ctx context.Context, id int64) *Error
+	ReplaceSuppliers(ctx context.Context, materialId int64, payload []MaterialSupplier) *Error
```
Comment (mnindrazaka) [inline apps/api/domain/material_repository.go]: Why we still need this, why not insert / replace the material suppliers on CreateMaterial and UpdateMaterialById repo ? so we hide repo implementation detail from the usecase. 

So in the future, if we migrate to no sql (for example), we can use the same repo, this ReplaceSuppliers separation only make sense on SQL style db, if we move to no sql, we can just send the material data with the suppliers

What do you think ?

File: apps/api/data/mysql/material_repo.go [hunk 4]
```diff
@@ -86,33 +86,61 @@ func (repo Repository) GetMaterialsWeeklyUsage(ctx context.Context, ids []int64)
...
 }
 
 func (repo Repository) CreateMaterial(ctx context.Context, material domain.Material) (domain.Material, *domain.Error) {
-	db := GetDbFromCtx(ctx, repo.db)
-	materialPayload := ToMaterialDB(material)
-	result := db.Table("materials").Create(&materialPayload)
-
-	if result.Error != nil {
-		return domain.Material{}, ToErrorCtx(ctx, result.Error, "CreateMaterial")
+	outerDb := GetDbFromCtx(ctx, repo.db)
```
Comment (mnindrazaka) [inline apps/api/data/mysql/material_repo.go]: can we keep the variable name db ? so it will consistent
```suggestion
	db := GetDbFromCtx(ctx, repo.db)
```

File: apps/api/data/mysql/material_repo.go [hunk 5]
```diff
@@ -86,33 +86,61 @@ func (repo Repository) GetMaterialsWeeklyUsage(ctx context.Context, ids []int64)
...
+		First(&created)
+	return ToMaterialDomain(created), ToErrorCtx(ctx, fetchResult.Error, "CreateMaterial")
 }
 
 func (repo Repository) UpdateMaterialById(ctx context.Context, material domain.Material, id int64) (domain.Material, *domain.Error) {
-	db := GetDbFromCtx(ctx, repo.db)
-	materialPayload := ToMaterialDB(material)
-	if result := db.Table("materials").Where("id = ?", id).Updates(&materialPayload); result.Error != nil {
-		return domain.Material{}, ToErrorCtx(ctx, result.Error, "UpdateMaterialById")
+	outerDb := GetDbFromCtx(ctx, repo.db)
```
Comment (mnindrazaka) [inline apps/api/data/mysql/material_repo.go]: this one also, keep it as db
```suggestion
	db := GetDbFromCtx(ctx, repo.db)
```
